File: Year 3/Fall Semester/CS-315_Programming_Languages/Homework1/Codes/gultekin_omeroktay.dart
foo(var key, var value) {
  print("Key: ${key}, Value: ${value}");
}

void main() {
  print("--------------------------- 1 ---------------------------");
  // Map<String, String> ListFinalAllInfos = {'stackoverflow': 'one', 'google': 'two'};
  var cs315Instructors = {
    "Section 1": "Altay Güvenir",
    "Section 2": "Aynur Dayanik",
    "Section 3": "Karani Kardas"
  };
  print("Initial CS 315 Instructors: ${cs315Instructors}");
  print("--------------------------- 1 ---------------------------");

  print("--------------------------- 2 ---------------------------");
  print(cs315Instructors["Section 3"]);
  print("--------------------------- 2 ---------------------------");

  print("--------------------------- 3 ---------------------------");
  cs315Instructors["Section 4"] = "Not Available";
  print(cs315Instructors);
  print("--------------------------- 3 ---------------------------");

  
  print("--------------------------- 4 ---------------------------");
  print(cs315Instructors.remove("Section 4"));
  print(cs315Instructors);
  print("--------------------------- 4 ---------------------------");
  
  print("--------------------------- 5 ---------------------------");
  cs315Instructors["Section 1"] = "Halil Altay Güvenir";
  print(cs315Instructors);
  print("--------------------------- 5 ---------------------------");
  
  print("--------------------------- 6 ---------------------------");
  print(cs315Instructors.containsKey("Section 2"));
  print(cs315Instructors.containsKey("Section 4"));
  print("--------------------------- 6 ---------------------------");

  print("--------------------------- 7 ---------------------------");
  print(cs315Instructors.containsValue("Karani Kardas"));
  print(cs315Instructors.containsValue("Karani"));
  print("--------------------------- 7 ---------------------------");

  print("--------------------------- 8 ---------------------------");
  cs315Instructors.forEach((k, v) => foo(k, v));
  print("--------------------------- 8 ---------------------------");
}

File: Year 3/Fall Semester/CS-315_Programming_Languages/Homework1/Codes/gultekin_omeroktay.rs
use std::collections::HashMap;

fn foo(k: &str, v: &str) {
    println!("{k}: {v}")
}

fn main() {
    println!("--------------------------- 1 ---------------------------");
    let mut cs315_instructors = HashMap::from([
        ("Section 1", "Altay Güvenir"),
        ("Section 2", "Aynur Dayanik"),
        ("Section 3", "Karani Kardas"),
    ]);

    // Error println!("{}", cs315_instructors);
    println!("{:?}", cs315_instructors);
    println!("--------------------------- 1 ---------------------------");

    println!("--------------------------- 2 ---------------------------");
    println!("{}", cs315_instructors["Section 3"]);
    println!("--------------------------- 2 ---------------------------");

    println!("--------------------------- 3 ---------------------------");
    cs315_instructors.insert("Section 4", "Not Available");

    println!("{:?}", cs315_instructors);
    println!("--------------------------- 3 ---------------------------");

    println!("--------------------------- 4 ---------------------------");
    cs315_instructors.remove("Section 4");

    println!("{:?}", cs315_instructors);
    println!("--------------------------- 4 ---------------------------");

    println!("--------------------------- 5 ---------------------------");
    cs315_instructors.insert("Section 1", "Halil Altay Güvenir");
    println!("{:?}", cs315_instructors);
    println!("--------------------------- 5 ---------------------------");

    println!("--------------------------- 6 ---------------------------");
    println!("{}", cs315_instructors.contains_key("Section 1"));
    println!("--------------------------- 6 ---------------------------");

    println!("--------------------------- 7 ---------------------------");
    println!( "{}", cs315_instructors.values() .any(|&val| val == "Karani Kardas"));
    println!("--------------------------- 7 ---------------------------");

    println!("--------------------------- 8 ---------------------------");
    for (key, value) in &cs315_instructors {
        foo(key, value);
    }
    println!("--------------------------- 8 ---------------------------");
}